src/Terrain/MapPerspective.java

```diff
@@ -194,24 +217,18 @@ public void resetFog()
       {
         for( Unit unit : co.units )
         {
-          for( XYCoord coord : Utils.findVisibleLocations(this, unit, false) )
-          {
-            revealFog(coord, false);
-          }
-          // We need to do a second pass with piercing vision so we can know whether to reveal the units
-          for( XYCoord coord : Utils.findVisibleLocations(this, unit, true) )
-          {
-            revealFog(coord, true);
-          }
+          UnitContext uc = new UnitContext(this, unit);
+          uc.calculateVision();
+          revealFog(uc, uc.coord);
         }
         for( XYCoord xyc : co.ownedProperties )
         {
           revealFog(xyc, true); // Properties can see themselves and anything on them
-          MapLocation loc = master.getLocation(xyc);
-          for( XYCoord coord : Utils.findVisibleLocations(this, loc.getCoordinates(), Environment.PROPERTY_VISION_RANGE) )
-          {
-            revealFog(coord, false);
-          }
+          if( isFogDoR() ) // Trilogy fog does not enable cities to have area vision
```

Comment:
It may make sense to move to a "What vision range do you have, here's the fog type" function call on the `TerrainType`, since Radars are a thing we could add at some point.
But I didn't feel like finagling that when I really just need 2 states and customizing terrains is annoying.